charts/argo-cd/templates/argocd-server/deployment.yaml

```diff
@@ -76,6 +76,10 @@ spec:
           {{- with (concat .Values.global.env .Values.server.env) }}
             {{- toYaml . | nindent 10 }}
           {{- end }}
+          {{- if not .Values.server.autoscaling.enabled }}
```

Comment:
This condition enables HPA that deploys between min and max amount of replicas, not a fixed number stated below.